crates/flox-rust-sdk/src/models/environment.rs

```diff
@@ -429,6 +440,20 @@ impl<S: TransactionState> PathEnvironment<S> {
     fn catalog_path(&self) -> PathBuf {
         self.path.join("pkgs").join("default").join(CATALOG_JSON)
     }
+
+    /// Attempt to transactionally replace the manifest contents
+    async fn transact_with_manifest_contents(
```

Comment:
Could we rename this function to...edit? Or are you anticipating edit and transact_with_manifest_contents will diverge in the future?

Reply:
The intention is to use this wherever we do transactions, so it feels confusing to call this method `edit`. That name also doesn't convey that it's doing transactions. I'm imagining that `edit`, `install`, and `uninstall` boil down to
- Get the current manifest
- Perform some operation on it
- Pass the new contents to `transact_with_manifest_contents`

Reply:
Because it's being called in the `Environment` impl, which is done for `S: TransactionState`.